nemo/core/classes/modelPT.py

```diff
@@ -1019,3 +1019,19 @@ def __is_model_being_restored() -> bool:
     def __set_model_restore_state(is_being_restored: bool):
         global _MODEL_IS_RESTORED
         _MODEL_IS_RESTORED = is_being_restored
+
+    def update_weights(self, checkpoint_path):
+        if os.path.exists(checkpoint_path):
+            raise FileExistsError(f"Could not find checkpoint file '{checkpoint_path}'!")
+        checkpoint = torch.load(checkpoint_path)
```

Comment:
If this is assumed to be a PTL checkpoint, then please simply do 

```python
checkpoint = torch.load(checkpoint_path)
model.load_state_dict(ckpt['state_dict'], strict=False)  # This strict should be a flag probably settable by function argument
del checkpoint  # this is necessary so that lingering checkpoint weights are not preserved in GPU memory
```

We do not want to silently log that certain weights were loaded and certain were not and let training continue as if everything is fine. If a user wants that, they must explicitly pass `strict=False` for such behaviour.

Reply:
@titu1994 @blisc do you want to drop it completely or keep it for now and then we can come up with a better solution done in exp_manager?

Reply:
Dropped it for now. Let's add a solution to exp_manager in another PR.